cordapp-example/workflows-kotlin/src/main/kotlin/com/example/flow/PaybackFlow.kt

```diff
@@ -0,0 +1,128 @@
+
+        /**
+         * The flow logic is encapsulated within the call() method.
+         */
+        @Suspendable
+        override fun call(): SignedTransaction {
+            // Obtain a reference to the notary we want to use.
+            val notary = serviceHub.networkMapCache.notaryIdentities[0]
+
+            // Stage 1.
+            progressTracker.currentStep = GENERATING_TRANSACTION
+            // Generate an unsigned transaction.
+            val txCommand = Command(IOUContract.Commands.Destroy(), ourIdentity.owningKey)
+
+            /* val criteria = QueryCriteria.VaultQueryCriteria(
+                    contractStateTypes = setOf(IOUState::class.java),
+                    status = Vault.StateStatus.UNCONSUMED,
+                    exactParticipants = listOf(ourIdentity, otherParty)
+            ) // */
+
+            //val signedTransaction = serviceHub.validatedTransactions.getTransaction(loanTxId)
+            // val signedTransaction = transactions.find { it.id == transactionHash } ?: throw IllegalArgumentException("Unknown transaction hash.")
+
+            val ourStateRef = StateRef(SecureHash.parse(loanTxId), 0)
+            // val (state, ref) = serviceHub.toStateAndRef<ContractState>(ourStateRef)
+            val inputStateAndRef = serviceHub.toStateAndRef<ContractState>(ourStateRef)
+            val inputIouState = inputStateAndRef.state.data as IOUState;
+
+            // build tx that will consume previous output
+            val txBuilder = TransactionBuilder(notary)
+                    .addInputState(inputStateAndRef)
+                    .addCommand(txCommand)
+
+            // Stage 2.
+            progressTracker.currentStep = VERIFYING_TRANSACTION
+            // Verify that the transaction is valid.
+            txBuilder.verify(serviceHub)
+
+            // Stage 3.
+            progressTracker.currentStep = SIGNING_TRANSACTION
+            // Sign the transaction.
+            val signedTx = serviceHub.signInitialTransaction(txBuilder, inputIouState.lender.owningKey)
+
+            /**
+             * I don't think lender sig is required here,
+             * we should be able to payback without it - but idk how to do `FinalityFlow` without it
+             * also, will it update lender vault and spend IUO if lender will nobe be included in flow?
+             */
+
+            // Send the state to the counterparty, and receive it back with their signature.
+            val otherPartySession = initiateFlow(inputIouState.lender)
+            //val fullySignedTx = subFlow(CollectSignaturesFlow(signedTx, setOf(otherPartySession), ExampleFlow.Initiator.Companion.GATHERING_SIGS.childProgressTracker()))
+
+            // Stage 4.
+            progressTracker.currentStep = FINALISING_TRANSACTION
+            // Notarise and record the transaction in both parties' vaults.
+            return subFlow(FinalityFlow(signedTx, setOf(otherPartySession), FINALISING_TRANSACTION.childProgressTracker()))
```

Comment:
I don't think lender sig is required here, we should be able to payback without it - but idk yet how to do `FinalityFlow` without it. Also, if I would remove lender signature will it update lender vault and spend IUO if lender will not be included in flow?